### PR TITLE
check if the db file path dir exists before setting up schema

### PR DESCRIPTION
**Pull request by theverything**
**What changed?**
This PR checks if the parent directories exist for the given database file path and returns a more descriptive error to the user.

**Why?**
The current error output is confusing. Closes https://github.com/temporalio/temporalite/issues/55.

**How did you test it?**
Yes. Tested it locally.

**Potential risks**
Users are confused by a new error?

**Is hotfix candidate?**
No.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -55,6 +56,12 @@ func NewServer(opts ...ServerOption) (*Server, error) {
 	if !c.Ephemeral {
 		// Apply migrations if file does not already exist
 		if _, err := os.Stat(c.DatabaseFilePath); os.IsNotExist(err) {
+			// Check if any of the parent dirs are missing
+			dir, _ := filepath.Split(c.DatabaseFilePath)
```
Comment (jlegrone) [inline server.go:60]: I think there's a bug here when starting temporalite with a state file in the current directory, eg. `temporalite start -f foo.db`. From the godoc:

> If there is no Separator in path, Split returns an empty dir and file set to path.

In this case, `filepath.Split` returns `"", "foo.db"`, and calling `os.Stat("")` returns an error.

Instead we could use [filepath.Dir](https://pkg.go.dev/path/filepath#Dir) which returns `"."` when the path is empty.

```suggestion
			dir := filepath.Dir(c.DatabaseFilePath)
```

Comment (theverything) [reply server.go]: I missed that part in the docs. Thank you.

**Review by jlegrone — APPROVED** [review]

**Comment by jlegrone** [comment]
LGTM, please just sign the CLA and we should be good to go. Thanks!

**Comment by theverything** [comment]
I agreed to the CLA a couple times but I'm not sure why it's not showing

**Comment by jlegrone** [comment]
Might be because your commits are signed off as @zillow-jeff, not @theverything.

**Review by cretz — APPROVED** [review]

**Pull request merged**